src/marshmallow/experimental/context.py

```diff
@@ -0,0 +1,26 @@
+"""Helper API for setting serialization/deserialization context."""
+
+import contextlib
+import contextvars
+import typing
+
+_T = typing.TypeVar("_T")
+_CURRENT_CONTEXT: contextvars.ContextVar = contextvars.ContextVar("context")
+
+
+class Context(contextlib.AbstractContextManager, typing.Generic[_T]):
+    def __init__(self, context: _T) -> None:
+        self.context = context
+        self.token: contextvars.Token | None = None
+
+    def __enter__(self) -> None:
+        self.token = _CURRENT_CONTEXT.set(self.context)
+
+    def __exit__(self, *args, **kwargs) -> None:
+        _CURRENT_CONTEXT.reset(typing.cast(contextvars.Token, self.token))
```

Comment:
you should be able to remove the cast if you check against None

```suggestion
        if self.token is not None:
            _CURRENT_CONTEXT.reset(self.token)
```

Reply:
I figured the cast was better because it is optimized out at compile time (right ?).

The perf impact is probably peanuts but as a principle, it kinda sucks to add a useless instruction to silent a linter.

I admit that the cast is less eye-candy, so I'm open to the change.

Reply:
that's fair. most of the time it's better to avoid casting for robustness. in this case, it can easily be reasoned that this line will never be reached unless `self.token` is set.